libvast/src/system/accountant.cpp

```diff
@@ -37,8 +40,75 @@
 #include <limits>
 
 namespace vast {
+
 namespace system {
 
+template <class V>
+bool assign(V& v, const caf::settings& m, std::string_view key) {
+  auto it = m.find(key);
+  if (it == m.end())
+    return true;
+  if constexpr (caf::detail::tl_contains<caf::config_value::variant_type::types,
+                                         V>::value) {
+    auto x = caf::get_if<V>(&it->second);
+    if (!x)
+      return false;
+    v = *x;
+    return true;
+  } else {
+    auto x = caf::get_if<std::string>(&it->second);
+    if (!x)
+      return false;
+    auto f = x->begin();
+    return parse(f, x->end(), v);
+  }
+}
+
+template <class V>
+bool assign(V& v, const map& m, const data& key) {
+  auto it = m.find(key);
+  if (it == m.end())
+    return true;
+  if constexpr (caf::detail::tl_contains<data::types, V>::value) {
+    auto x = caf::get_if<V>(&it->second);
+    if (!x)
+      return false;
+    v = *x;
+    return true;
+  } else {
+    auto x = caf::get_if<std::string>(&it->second);
+    if (!x)
+      return false;
+    auto f = x->begin();
+    return parse(f, x->end(), v);
+  }
+}
+
+caf::expected<accountant_config>
+to_accountant_config(const caf::settings& opts) {
+  accountant_config result;
+  assign(result.enable, opts, "enable");
+  assign(result.enable_self_sink, opts, "enable_self_sink");
+  assign(result.file_sink.enable, opts, "file_sink.enable");
+  assign(result.file_sink.path, opts, "file_sink.path");
+  assign(result.uds_sink.enable, opts, "uds_sink.enable");
+  assign(result.uds_sink.path, opts, "uds_sink.path");
+  assign(result.uds_sink.type, opts, "uds_sink.type");
+  return result;
+}
+
+caf::expected<accountant_config> to_accountant_config(const map& opts) {
+  accountant_config result;
+  assign(result.enable, opts, "enable");
+  assign(result.enable_self_sink, opts, "enable_self_sink");
+  assign(result.file_sink.enable, opts, "file_sink.enable");
+  assign(result.file_sink.path, opts, "file_sink.path");
+  assign(result.uds_sink.enable, opts, "uds_sink.enable");
+  assign(result.uds_sink.path, opts, "uds_sink.path");
+  assign(result.uds_sink.type, opts, "uds_sink.type");
+  return result;
+}
```

Comment:
```suggestion
// somewhere internal
template <class T>
caf::expected<accountant_config>
to_accountant_config_impl(const T& opts) {
  accountant_config result;
  assign(result.enable, opts, "enable");
  assign(result.enable_self_sink, opts, "enable_self_sink");
  assign(result.file_sink.enable, opts, "file_sink.enable");
  assign(result.file_sink.path, opts, "file_sink.path");
  assign(result.uds_sink.enable, opts, "uds_sink.enable");
  assign(result.uds_sink.path, opts, "uds_sink.path");
  assign(result.uds_sink.type, opts, "uds_sink.type");
  return result;
}


caf::expected<accountant_config>
to_accountant_config(const caf::settings& opts) {
  return to_accountant_config_impl(opts);
}

caf::expected<accountant_config> to_accountant_config(const map& opts) {
  return to_accountant_config_impl(opts);
}
```

Reply:
We don't need the `vast::data` overload yet, so I removed it for now.